OloEngine/src/OloEngine/Renderer/Commands/FrameDataBuffer.cpp

```
@@ -0,0 +1,163 @@
+    const glm::mat4* FrameDataBuffer::GetTransformPtr(u32 offset) const
+    {
+        std::lock_guard<std::mutex> lock(m_TransformMutex);
+        if (offset >= m_Transforms.size())
+        {
+            OLO_CORE_ERROR("FrameDataBuffer: Invalid transform offset {}", offset);
+            return nullptr;
+        }
+        return &m_Transforms[offset];
+    }
+
+    void FrameDataBuffer::WriteBoneMatrices(u32 offset, const glm::mat4* data, u32 count)
```

Clang (14.0.6)

[warning] method 'WriteBoneMatrices' can be made static

(readability-convert-member-functions-to-static)